articles/js-execution-under-the-hood/index.md

```diff
@@ -0,0 +1,167 @@
+3.     const result = inputNumber*2;
+4.     return result;
+5.     }
+6. const output = multiplyBy2(4);
+```
+
+As you execute functions, you will switch Execution Contexts. From Global to Local, and back to Global.
+
+This switching of Execution Contexts raises, in JavaScript, an issue of tracking where one is in their Thread Of Execution.
+
+To keep track of this, JavaScript uses a [**Call Stack**](https://developer.mozilla.org/en-US/docs/Glossary/Call_stack).
+
+And so now, you have a third element to the execution of your JavaScript code. The image below illustrates the three elements before any code is executed.
+
+![Call Stack illustration](call_stack_intro.jpg)
+
+Having understood that, you now get to see exactly how the call stack comes into play here.
+
+Line 1: Save in memory a constant, **num**, as the label for an integer, **3**.
+
+Line 2: Save in memory **multiplyBy2**, as the label for the functionality of the function, `multiplyBy2()`.
+
+Line 6: Save in memory a constant, **output**, as the label for whatever the result of the function  `multiplyBy2()` returns after executing.
+
+At this point, you call  `multiplyBy2()`.
+
+This adds `multiplyBy2()` to the call stack. And creates a Local Execution Context for the execution of the code inside  `multiplyBy2()`.As illustrated in the image below.
+
+![Add multiplyBy2() to call stack](call_stack_add_multiplyBy2.jpg)
+
+Once inside the Local Execution Context, you carry on with the two parts of an Execution context.
+
+The code executed in this case is the body of `multiplyBy2()`. Since, in JavaScript, you go into the body of a function when it is called.
+
+Line 3: Save in memory a constant, **result**, as the label for the product of the argument **inputNumber** and an integer **2**, an integer, **8**.
+
+Line 4: The [**return**](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/return) keyword then ends function execution and specifies a value to be returned to the function caller.
+
+![Local Execution Context](local_execution_context.jpg)
+
+You then return to the Global Execution Context. A few things happen when you do:
+
+1. JavaScript [**Garbage Collects**](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Memory_Management) the Local Execution Context's Variable Environment (Memory).
+2. The Call Stack is cleared.
+3. The return value is assigned to the constant **output**.
+
+The image below illustrates the Global Execution Context after all this is done.
+
+![After Garbage Collection](after_garbage_collection.jpg)
+
+Once all the lines of code are executed, JavaScript garbage collects the Global Execution Context's Variable Environment (Memory). 
+
+This brings you back to where you started as illustrated below.
+
+![Global Execution Context after you execute js code](global_execution_context_after_executing_js_code.jpg)
+
+Congratulations! You now have the full circle of what happens under the hood when you execute your JavaScript Code.
```

Comment:
You may want to say, "You have now seen the entire circle of what happens" instead of, "You now have the full circle of what happens"